libraries/chain/db_block.cpp

```diff
@@ -103,6 +140,32 @@ std::vector<block_id_type> database::get_block_ids_on_fork(block_id_type head_of
   result.emplace_back(branches.first.back()->previous_id());
   return result;
 }
+    
+void database::check_tansaction_for_duplicated_operations(const signed_transaction& trx)
+{
+   const auto& proposal_index = get_index<proposal_object>();
+   std::set<fc::sha256> existed_operations_digests;
+   
+   proposal_index.inspect_all_objects( [&](const object& obj){
+      const proposal_object& proposal = static_cast<const proposal_object&>(obj);
+      for (auto& operation: proposal.proposed_transaction.operations)
+      {
+         existed_operations_digests.insert(fc::digest(operation));
+      }
+   });
+   
+   for (auto& pending_transaction: _pending_tx)
```

Comment:
This is not necessary since the effect of pending transactions is always visible in current chain state, at least from the PoV of the API.

Reply:
That's very good to know, thank you.